frontend/src/pages/news/insideNewsPage.jsx

```diff
@@ -34,14 +37,156 @@ export default function InsideNewsPage() {
         if(article) setSections(Object.values(article?.article))
     }, [article]);
 
+    const [likeStatus, setLikeStatus] = useState(false);
+    const [initialLikeStatus, setInitialLikeStatus] = useState(false);
+    const elementRef = useRef(null);
+    const [hasRead, setHasRead] = useState(false);
+    const likeStatusRef = useRef(likeStatus);
+    const initialLikeStatusRef = useRef(initialLikeStatus);
+
+    const { user } = useContext(AuthContext)
+
+
+    useEffect(() => {
+        const observer = new IntersectionObserver(
+            ([entry]) => {
+                if (entry.isIntersecting && !hasRead) {
+                    saveInteraction('read');
+                    setHasRead(true);
+                }
+            },
+            { threshold: 0.8 }
+        );
+
+        if (elementRef.current) {
+            observer.observe(elementRef.current);
+        }
+
+        return () => {
+            if (elementRef.current) {
+                observer.unobserve(elementRef.current);
+            }
+        };
+    }, [hasRead]);
+
+    useEffect(() => {
+        const handleEvent = () => {
+            handleLikeStatus();
+        };
+
+        window.addEventListener("beforeunload", handleEvent);
+        window.addEventListener("popstate", handleEvent);
+
+        return () => {
+            window.removeEventListener("beforeunload", handleEvent);
+            window.removeEventListener("popstate", handleEvent);
+        };
+    }, []);
+
+    useEffect(() => {
+        initialLikeStatusRef.current = initialLikeStatus;
+    }, [initialLikeStatus]);
+
+    useEffect(() => {
+        likeStatusRef.current = likeStatus;
+    }, [likeStatus]);
+
+    const handleLikeStatus = async () => {
+        if (likeStatusRef.current !== initialLikeStatusRef.current) {
+            let interactionType = likeStatusRef.current ? 'like' : 'dislike';
+            saveInteraction(interactionType);
+        }
+    };
+
+    useEffect(() => {
+        const logInteraction = async () => {
+            await saveInteraction('open');
+        };
+        logInteraction();
+    }, []);
+
+    const saveInteraction = async (interactionType) => {
+        if (user) {
+            try {
+                await axios.post(
+                    `${apiEndpoints.url}${apiEndpoints.interactions.saveInteraction}`,
+                    {
+                        user_id: user.id,
+                        blob_id: articleId,
+                        interaction_type: interactionType,
+                    }
+                );
+            } catch (error) {
+                toast.error(`Troubles with saving interaction: ${error}`)
+            }
+        }
+    };
+
+    useEffect(() => {
+        async function getInteractionsCounts() {
+            try {
+                const { data } = await axios.get(`${apiEndpoints.url}${apiEndpoints.interactions.getInteractionsCounts}`, {
+                    params: {
+                        blob_id: articleId,
```

Comment:
I don't particularly like `blob_id`. I do realize what that means, but since we only store articles in our blob (as far as I know that is) maybe we could go with something like "article_blob_id" at least?